Formula/i/include-what-you-use.rb

```diff
@@ -96,6 +95,7 @@ def install
 
     (testpath/"main.cc").write <<~CPP
       #include <iostream>
+      #include <__ostream/basic_ostream.h>
```

Comment:
This is an internal header -- you shouldn't be `#include`-ing this.

Reply:
agree, this is more of an upstream issue at this point

Reply:
IWYU maintainer here. Our Mac support is actually really limited, because we don't have Mac users among the maintainers. I happen to have a Mac for work, but it's not my main environment.

If you're keeping your own test suite, I wonder if it would be better to use an isolated example that doesn't depend on system headers...?

I managed to patch the IWYU test suite to run on my Mac (prefix the `include-what-you-use` invocations with `xcrun`), and I see 7 failures out of the box, presumably caused by differences between libstdc++ and libc++. You might also consider patching them to be marked `XFAIL` and then just run the actual test suite.

Reply:
I see libc++ from llvm-19 ships with generated IWYU mappings. You could try adding `-Xiwyu -mapping_file=.../libcxx.imp` where `...` is the path to the libc++ include root (not sure how to best compute that for a brew environment).

Reply:
yeah, I can give that a try and see. 👍

Reply:
adding the mapping_file works for me.

Reply:
maybe a newbie question, why is llvm mapping file not in the search path?

Reply:
What is the default path to the mapping file? Can we configure that somehow?

Reply:
> maybe a newbie question, why is llvm mapping file not in the search path?
> What is the default path to the mapping file? Can we configure that somehow?

The header search path is not used for mappings, and there's no separate explicit search path for them. The simple search rule is described in this section of the docs: https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUMappings.md#command-line-switches-for-mapping-files

That said, we have an idea to try and probe .imp files from the normal header search path: https://github.com/include-what-you-use/include-what-you-use/issues/1271, but there's quite a few small things in the way to make it happen.

Reply:
Thanks! I wonder if we can update `iwyu_include_picker.cc` (perhaps using one of your `.py` scripts) somehow for usage with `libc++`.

Reply:
The scripts in https://github.com/include-what-you-use/include-what-you-use/tree/master/mapgen are per-library, there's nothing generic about them. We don't have one for libc++, so there's not much to gain there.

The libcxx.imp provided by libc++ is generated by a script in the libc++ tree: https://github.com/llvm/llvm-project/blob/main/libcxx/utils/generate_iwyu_mapping.py -- maybe you could use that to

* generate an .imp file
* rewrite/transpile to C++
* patch into `iwyu_include_picker.cc` to overwrite `libcxx_include_map`

But it sounds a little messy.